src/fields/Fields.cpp

```diff
@@ -365,27 +367,46 @@ Fields::InterpolateBoundaries (amrex::Vector<amrex::Geometry> const& geom, const
             {
                 if (i==nx_fine_low || i== nx_fine_high || j==ny_fine_low || j == ny_fine_high) {
                     // Compute coordinate on fine grid
-                    const amrex::Real x = plo[0] + (i+lo[0]+0.5_rt)*dx[0];
-                    const amrex::Real y = plo[1] + (j+lo[1]+0.5_rt)*dx[1];
-                    // index left (in x) and below (in y) of the compute coordinate on coarse grid
-                    const int idx_left = (i + lo[0]) / refinement_ratio[0];
-                    const int idx_down = (j + lo[1]) / refinement_ratio[1];
-                    const amrex::Real x_left = plo_coarse[0]+(idx_left +0.5_rt)*dx_coarse[0];
-                    const amrex::Real y_down = plo_coarse[1]+(idx_down +0.5_rt)*dx_coarse[1];
-
-                    // Bilinear interpolation from coarse to fine grid
-                    const amrex::Real val_left_down  = data_array_coarse(idx_left  , idx_down  ,iz);
-                    const amrex::Real val_left_up    = data_array_coarse(idx_left  , idx_down+1,iz);
-                    const amrex::Real val_right_up   = data_array_coarse(idx_left+1, idx_down+1,iz);
-                    const amrex::Real val_right_down = data_array_coarse(idx_left+1, idx_down  ,iz);
-                    const amrex::Real df_x = val_right_down - val_left_down;
-                    const amrex::Real df_y = val_left_up - val_left_down;
-                    const amrex::Real df_xy = val_left_down + val_right_up - val_right_down
-                                             -val_left_up;
-
-                    const amrex::Real boundary_value =
-                        df_x*(x-x_left)/dx_coarse[0] + df_y*(y-y_down)/dx_coarse[1] +
-                        df_xy*(x-x_left)*(y-y_down)/(dx_coarse[0]*dx_coarse[1]) + val_left_down;
+                    amrex::Real x, y;
+                    if (i==nx_fine_low) {
+                        x = plo[0] + (i+lo[0]+nguards_xy-0.5_rt)*dx[0];
+                    } else if (i== nx_fine_high) {
+                        x = plo[0] + (i+lo[0]+nguards_xy+1.5_rt)*dx[0];
+                    } else {
+                        x = plo[0] + (i+lo[0]+nguards_xy+0.5_rt)*dx[0];
+                    }
```

Comment:
Could you add some comments to explain these values? Why always add guard cells? Why -0.5/+1.5? In particular, what puzzled me is that:
```
        i                                x
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
nx_fine_low      plo[0] + (nx_fine_low+lo[0]+nguards_xy-0.5_rt)*dx[0];
nx_fine_low+1    plo[0] + (nx_fine_low+lo[0]+nguards_xy+1.5_rt)*dx[0]; # the one above + 2*dx
nx_fine_low+2    plo[0] + (nx_fine_low+lo[0]+nguards_xy+2.5_rt)*dx[0]; # the one above + 1*dx
nx_fine_low+3    plo[0] + (nx_fine_low+lo[0]+nguards_xy+3.5_rt)*dx[0]; # the one above + 1*dx
etc.
```
Is this expected? Same question around `nx_fine_high`, and for the y direction.